### PR TITLE
Auto-generate release notes also from build tags + Miscellaneous chores

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

**Comment by lluiscampos** [comment]
@kacf @oleorhagen The intention is to get the rid of this:

image

File: extra/release_tool.py [hunk 1]
```diff
@@ -2849,7 +2849,19 @@ def do_release(release_state_file):
         elif reply.lower() == "l":
             do_license_generation(state, tag_avail)
         elif reply.lower() == "n":
-            do_generate_release_notes(state["repo_dir"], state["version"])
+            # Generate release notes for final tag if exists. Otherwise for the current build tag
+            tag_release_notes = state["version"]
+            try:
+                execute_git(
+                    state,
+                    repo.git(),
```
Comment (kacf) [inline extra/release_tool.py:2857]: Is `repo` valid here..?

File: extra/release_tool.py [hunk 2]
```diff
@@ -2849,7 +2849,19 @@ def do_release(release_state_file):
...
+                    ["rev-parse", state["version"]],
+                    capture=True,
+                    capture_stderr=True,
+                )
```
Comment (kacf) [inline extra/release_tool.py:2861]: I think you don't need this, the information should already be available in `tag_avail`. I'm a bit rusty nowadays, but I think it should be `tag_avail["integration"]["build_tag"]`, which should always be the latest build tag, or the final tag, if it exists (check description of `check_tag_availability`).

File: extra/release_tool.py [hunk 3]
```diff
@@ -2849,7 +2849,19 @@ def do_release(release_state_file):
         elif reply.lower() == "l":
             do_license_generation(state, tag_avail)
         elif reply.lower() == "n":
```
Comment (kacf) [inline extra/release_tool.py:2851]: Please also change the description of the `n` option.

File: extra/release_tool.py (hunk 2 above)
Comment (lluiscampos) [reply extra/release_tool.py:2861]: Thanks. I actually tried to find this exact path (that `tag_avail["integration"]["build_tag"]` will be updated with the final tag) but I couldn't see it. I'll look more into it :eyes:

**Review by oleorhagen — APPROVED** [review]
Nice!

Not too familiar with this code though.

File: extra/release_tool.py (hunk 1 above)
Comment (lluiscampos) [reply extra/release_tool.py]: This comment is now obsolete.

**Comment by lluiscampos** [comment]
@mender-test-bot cherry-pick to:
* 3.5.x

**Comment by lluiscampos** [comment]
@mender-test-bot cherry-pick to:
* 3.3.x

**Pull request merged**